src/mol-plugin-ui/structure/measurements.tsx

```diff
@@ -80,13 +87,32 @@ export class MeasurementControls extends PurePluginUIComponent<{}, { isBusy: boo
     componentDidMount() {
         this.subscribe(this.selection.events.additionsHistoryUpdated, () => {
             this.forceUpdate();
+            this.updateOrderLabels();
         });
 
         this.subscribe(this.plugin.behaviors.state.isBusy, v => {
             this.setState({ isBusy: v });
         });
     }
 
+    componentDidUpdate(prevProps: {}, prevState: { isBusy: boolean, action?: 'add' | 'options' }) {
+        if (this.state.action !== prevState.action)
+            this.updateOrderLabels();
+    }
```

Comment:
I think this will keep showing the labels if you hide the whole panel by clicking on the header. Could you please handle that case too?

Reply:
Thanks, I didn't think of that. Fixed.